src/common/initializable-state/create.ts

```diff
@@ -5,11 +5,13 @@
 
 import type { DiContainerForInjection, Injectable, InjectionToken } from "@ogre-tools/injectable";
 import { getInjectable } from "@ogre-tools/injectable";
+import type { Runnable } from "../runnable/run-many-for";
 
 export interface CreateInitializableStateArgs<T> {
```

Comment:
Could `InitializableState` and "`TemporallyInitializableState`" (utilising the `when`) be two different responsibilities?

Reply:
Maybe in the future, but I have not encountered a use case for the second.